core/account/utxodb/reserve.go

```diff
@@ -57,15 +58,23 @@ type (
 	}
 )
 
+func beginTxn(ctx context.Context) (pg.Tx, error) {
+	db := pg.FromContext(ctx)
+	beginner, ok := db.(pg.Beginner)
```

Comment:
I think maybe we should type assert to the concrete type `*sql.DB` inline at the call sites.